src/main/java/org/example/controller/TraineeController.java

```diff
@@ -160,13 +166,16 @@ public ResponseEntity<Void> activateTrainee(@RequestBody ActivateUserRequestDto
         }
     }
 
+
     @Operation(summary = "Deactivate a trainee", description = "Deactivate a trainee account", tags = { "Trainee" })
     @ApiResponses(value = {
             @ApiResponse(responseCode = "200", description = "Trainee deactivated successfully"),
             @ApiResponse(responseCode = "500", description = "Internal server error")
     })
     @PatchMapping("/deactivate")
-    public ResponseEntity<Void> deactivateTrainee(@RequestBody DeactivateUserRequestDto deactivateUserRequestDto) {
+    public ResponseEntity<Void> deactivateTrainee(
+            @AuthenticationPrincipal UserDetails userDetails,
+            @RequestBody DeactivateUserRequestDto deactivateUserRequestDto) {
         try {
```

Comment:
We should not catch and process exceptions like this in controller. We should exception handler. For example using @RestControllerAdvice and @ExceptionHandler